File: src/main/java/edu/cornell/cs/apl/viaduct/imp/ast/values/BooleanValue.java
package edu.cornell.cs.apl.viaduct.imp.ast.values;

import com.google.auto.value.AutoValue;
import edu.cornell.cs.apl.viaduct.imp.ast.types.BooleanType;
import edu.cornell.cs.apl.viaduct.imp.ast.types.ImpBaseType;

/** Boolean literal. */
@AutoValue
public abstract class BooleanValue implements ImpValue {
  public static BooleanValue create(boolean value) {
    return new AutoValue_BooleanValue(value);
  }

  public abstract boolean getValue();

  @Override
  public final ImpBaseType getType() {
    return BooleanType.create();
  }

  @Override
  public final String toString() {
    return Boolean.toString(this.getValue());
  }
}

File: src/main/kotlin/edu/cornell/cs/apl/viaduct/imp/values/BooleanValue.kt
package edu.cornell.cs.apl.viaduct.imp.values

import edu.cornell.cs.apl.viaduct.imp.types.BooleanType
import edu.cornell.cs.apl.viaduct.imp.types.ValueType

/** A boolean. */
data class BooleanValue(val value: Boolean) : Value {
    override val type: ValueType
        get() = BooleanType

    override fun toString(): String {
        return value.toString()
    }
}
